moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -503,6 +504,19 @@ void PlanningScene::checkCollision(const collision_detection::CollisionRequest&
                                    const moveit::core::RobotState& robot_state,
                                    const collision_detection::AllowedCollisionMatrix& acm) const
 {
+  // lock the octomap if there is any as it might be shared with other PlanningScenes
+  collision_detection::OccMapTree::ReadLock lock;
+  collision_detection::CollisionEnv::ObjectConstPtr map = world_->getObject(OCTOMAP_NS);
+  if (map)
+  {
+    if (map->shapes_.size() == 1)
+    {
```

Comment:
```suggestion
  if (map && map->shapes_.size() == 1)
  {
```
`&&` is a [sequence point](https://stackoverflow.com/questions/4176328/what-are-sequence-points-and-how-do-they-affect-undefined-behavior), no need for the nested testing.